src/components/Modals/ContactFormModal.jsx

```diff
@@ -27,18 +58,9 @@ const ContactFormModal = ({
       <ContactForm
         contact={contact}
         onSubmit={async newContact => {
-          let resp
-          if (contact) {
-            const payload = {
-              _type: contact._type,
-              _id: contact._id,
-              _rev: contact._rev,
-              ...newContact
-            }
-            resp = await updateContact(payload)
-          } else {
-            resp = await createContact(newContact)
-          }
+          const createOrUpdate = contact ? updateContact : createContact
+          const updatedContact = getUpdatedContact(contact, newContact)
+          const resp = await createOrUpdate(updatedContact)
```

Comment:
Can we dispatch an action ? And having async stuff inly in our thunk?

Reply:
No try / catch ?

Reply:
if we can avoid redux I think we should